### PR TITLE
fix: update static url for image and link

### PR DESCRIPTION
**Pull request by leangseu-edx**
**TL;DR -** correctly update static url for image and link

JIRA: [AU-490](https://openedx.atlassian.net/browse/AU-490)

**What changed?**

- Modify **static** url on insert/edit image and anchor link (update logic)
- Modify **static** url on save contents (getcontent in this case)
- Remove **Sanitize** url on init because it would be redundant with on save.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

Follow the jira ticket instruction

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/xblock/static/js/src/studio/oa_tiny_mce.js [hunk 1]
```diff
@@ -23,30 +23,46 @@ export const oaTinyMCE = (options) => {
 
   const getFonts = () => CUSTOM_FONTS + STANDARD_FONTS;
   const baseAssetUrl = options.base_asset_url;
+  const staticUrl = '/static';
+
+  const sanitizeContent = (content) => {
+    // rewrite anchor link
+    let newContent = rewriteStaticLinks(
```
[the remark this answers is not in the document]
Comment (leangseu-edx) [reply openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:30]: `edx-platform` a global js function

[the remark this answers is not in the document]
Comment (matthugs) [reply openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:30]: looks like here: https://github.com/openedx/edx-platform/blob/db32ff2cdf678fa8edd12c9da76a76eef0478614/common/static/js/src/utility.js#L21

File: openassessment/xblock/static/js/src/studio/oa_tiny_mce.js [hunk 2]
```diff
@@ -23,30 +23,46 @@ export const oaTinyMCE = (options) => {
...
+      content,
+      `<a href="${staticUrl}`,
+      `<a href="${baseAssetUrl}`,
+    );
+
+    // rewrite image src
+    newContent = rewriteStaticLinks(
+      newContent,
+      `<img src="${staticUrl}`,
+      `<img src="${baseAssetUrl}`,
```
Comment (matthugs) [inline openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:40]: This is a bit of a hack, which would fail on html like, say ``

Why not use this the same way we do elsewhere in this library? https://github.com/openedx/edx-ora2/blob/dd6adeaf0572ecff3491e487f1df8eb0e1595c6f/openassessment/xblock/static/js/src/studio/oa_tiny_mce.js#L42

File: openassessment/xblock/static/js/src/studio/oa_tiny_mce.js [hunk 3]
```diff
@@ -23,30 +23,46 @@ export const oaTinyMCE = (options) => {
...
+    if (data[key]) {
+      data[key] = rewriteStaticLinks(data[key], ...urls);
     }
   };
 
-  const setupTinyMCE = (ed) => {
-    ed.on('SaveImage', dataHandler('src', baseAssetUrl, '/static'));
-    ed.on('EditImage', dataHandler('src', '/static', baseAssetUrl));
-    ed.on('SaveLink', dataHandler('html', baseAssetUrl, '/static'));
-    ed.on('EditLink', dataHandler('html', '/static', baseAssetUrl));
```
Comment (matthugs) [inline openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:38]: It looks like the original code wanted to translate from one format in one direction and to that same format in the other direction. Did we go about doing that incorrectly, before?

Said another way: how does this code behave in the following user story?
As a course author, I expect that when I write certain text into the platform and save it, then close the tab and return to edit it later, I find the same text I entered is still there.

File: openassessment/xblock/static/js/src/studio/oa_tiny_mce.js (hunk 2 above)
Comment (leangseu-edx) [reply openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:40]: getContent is actually get everything in html. If the user put anything even text that has `/static/` will get replace. It can be quite a confusing experience. This is just an improvement to what it was.

For image tag, it always generate ``. I tested with a few iteration of input.

File: openassessment/xblock/static/js/src/studio/oa_tiny_mce.js (hunk 3 above)
Comment (leangseu-edx) [reply openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:38]: I was a bit confuse too. As far as I can tell, `SaveLink` and `EditLink` was added later. The `dataHandler` was written to work for image only. There is slight change I put in.
For how it ended up in reverse order for one of them, I am really not sure. I tested with `debugger` on, it doesn't seem to have a correct behavior.

Current behavior, if you input `/static/...` in the url, it will get transform immediately. I do not have opinion on when it should get transform. I was just trying to make work from the previous code.

File: openassessment/xblock/static/js/src/studio/oa_tiny_mce.js (hunk 2 above)
Comment (matthugs) [reply openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:40]: Ok, so it sounds like we only support the rewriting of static links if folks insert the image via the WYSIWYG widget for adding images, and we don't rewrite static links if the user types html manually. Is that right?

If I have that right, I think a comment explaining what you've explained here would be helpful. (I would also urge you to make the comment longer than you'd consider necessary.)

File: openassessment/xblock/static/js/src/studio/oa_tiny_mce.js (hunk 3 above)
Comment (matthugs) [reply openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:38]: @leangseu-edx could you answer directly my question about the user story?

Comment (leangseu-edx) [reply openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:38]: When the user write text, it should return the same contents he entered.
If he input images or anchor, there will be modification if and only if the user are copying from `studio -> File & Upload -> Copy Urls for studio`. It has the specific format of `/static/....`.

**Review by matthugs — APPROVED** [review]

**Review by jansenk — APPROVED** [review]

File: openassessment/xblock/static/js/src/studio/oa_tiny_mce.js [hunk 4]
```diff
@@ -23,30 +23,57 @@ export const oaTinyMCE = (options) => {
...
+   * Within Studio -> File & Uploads -> Copy Urls.
+   * If the user used copy studios, the link get will be formatted
+   * similiar to `/static/....`. This will be invalid for usage and we
+   * need to modify it to be a correct link.
+   * @param {string} content HTML String
+   * @returns {string} new content
+   */
+  const rewriteContentUrl = (content) => {
+    // These url are generated by WYSIWYG editor, it is assumed to be formatted
+    // <a href='.....' >...</a> for link.
```
Comment (matthugs) [inline openassessment/xblock/static/js/src/studio/oa_tiny_mce.js:39]: I think a future reader of this code would be able to read the lines below it and know that this code would only work for things formatted as in the example in this comment. So they can probably figure out what your comment tells them, even without you having commented it.

The part that is most non-obvious about this code is why the assumption that anchor tags (and image tags, below) can safely be assumed to be in that format. Why that is is probably the most important thing to include in a comment here.
